### PR TITLE
StateGraphRecovery: Loading parameters from configuration files

### PR DESCRIPTION
**Pull request by Bonnie1256**

### CODE REVIEW

File: tests/test_stategraph.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import claripy
+import angr
+from angr.analyses.state_graph_recovery import MinDelayBaseRule, RuleVerifier, IllegalNodeBaseRule
+
+
+binaries_base = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'binaries')
+
+
+def test_smoketest():
```
Comment (cq674350529) [inline tests/test_stategraph.py:12]: The path seems not in the `binaries` directory ???

File: tests/test_stategraph.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    path = r"C:\Users\Fish\Desktop\temp\mitre\Traffic_Light_Short_Ped\build\Traffic_Light_Short_Ped.so"
```
Comment (cq674350529) [inline tests/test_stategraph.py:13]: The path seems not in the `binaries` directory ???

Comment (ltfish) [reply tests/test_stategraph.py:13]: This branch will not be merged to master any time soon. I will make sure to move this binary to the binaries repo before merging to master. Thanks for pointing out!

File: angr/analyses/state_graph_recovery/state_graph_recovery.py [hunk 3]
```diff
@@ -137,10 +141,25 @@ def _traverse_one(self, state: 'SimState'):
...
+
+
             s = simgr.active[0]
 
             simgr.stash(lambda x: x.addr == self._ret_trap, from_stash='active', to_stash='finished')
 
+
+
+
+
```
Comment (ltfish) [inline angr/analyses/state_graph_recovery/state_graph_recovery.py:162]: Let's remove the redundant blank lines here.

**Review by ltfish — CHANGES_REQUESTED** [review]

File: angr/analyses/state_graph_recovery/state_graph_recovery.py [hunk 4]
```diff
@@ -17,12 +17,15 @@ class StateGraphRecoveryAnalysis(Analysis):
     """
     Traverses a function and derive a state graph with respect to given variables.
     """
-    def __init__(self, func: 'Function', fields: 'AbstractStateFields', init_state: Optional['SimState']=None):
+    def __init__(self, func: 'Function', fields: 'AbstractStateFields', time_addr, init_state: Optional['SimState']=None):
```
Comment (ltfish) [inline angr/analyses/state_graph_recovery/state_graph_recovery.py:20]: What's the type of `time_addr`? Type-annotate it please.

**Review by ltfish — CHANGES_REQUESTED** [review]

File: angr/analyses/state_graph_recovery/state_graph_recovery.py [hunk 5]
```diff
@@ -17,12 +17,15 @@ class StateGraphRecoveryAnalysis(Analysis):
     """
     Traverses a function and derive a state graph with respect to given variables.
     """
-    def __init__(self, func: 'Function', fields: 'AbstractStateFields', init_state: Optional['SimState']=None):
+    def __init__(self, func: 'Function', fields: 'AbstractStateFields', time_addr: 'int', init_state: Optional['SimState']=None):
```
Comment (ltfish) [inline angr/analyses/state_graph_recovery/state_graph_recovery.py:20]: Your type annotation is wrong. Remove the quotes around `int`.

**Pull request merged**